cg/meta/rsync/rsync_api.py

```diff
@@ -140,10 +140,50 @@ def create_log_dir(self, dry_run: bool) -> None:
         else:
             log_dir.mkdir(parents=True, exist_ok=True)
 
+    def slurm_rsync_single_case(
```

Comment:
Consider breaking this function into smaller functions. If that is done and doc strings are added to those functions it would be much easier to understand what is going on.

Reply:
Good point. I'll see what I can break out!